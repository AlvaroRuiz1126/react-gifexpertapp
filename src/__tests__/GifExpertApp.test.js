import React from 'react';
import GifExpertApp from '../GiftExpertApp';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

describe('Pruebas del componente GifExpertApp', () => {
    test('Hace match con el snapshot del componente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar una lista de categorÃ­as', () => {
        const categories = ['naruto', 'Dragon ball z'];
        const wrapper = shallow(<GifExpertApp defaultCategory={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
    
})
