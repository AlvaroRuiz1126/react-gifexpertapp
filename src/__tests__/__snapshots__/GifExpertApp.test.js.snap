// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifExpertApp Debe mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categories="naruto"
      key="naruto"
    />
    <GifGrid
      categories="Dragon ball z"
      key="Dragon ball z"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas del componente GifExpertApp Hace match con el snapshot del componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
